snare/tests/test_get_body.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+import aiohttp
+import yarl
+import sys
+import os
+import shutil
+import asyncio
+from snare.cloner import Cloner
+from snare.utils.asyncmock import AsyncMock
+from snare.utils.page_path_generator import generate_unique_path
+
+
+class TestGetBody(unittest.TestCase):
+    def setUp(self):
+        self.main_page_path = generate_unique_path()
+        os.makedirs(self.main_page_path)
+        self.root = 'http://example.com'
+        self.level = 0
+        self.max_depth = sys.maxsize
+        self.loop = asyncio.new_event_loop()
+        self.css_validate = 'false'
+        self.handler = Cloner(self.root, self.max_depth, self.css_validate)
+        self.target_path = '/opt/snare/pages/{}'.format(yarl.URL(self.root).host)
+        self.return_content = None
+        self.expected_content = None
+        self.filename = None
+        self.hashname = None
+        self.url = None
+        self.content = b'''<html><body><a href="http://example.com/test"></a></body></html>'''
+
+        self.session = aiohttp.ClientSession
+        self.session.get = AsyncMock(
+            return_value=aiohttp.ClientResponse(
+                url=yarl.URL("http://www.example.com"), method="GET", writer=None, continue100=1,
+                timer=None, request_info=None, traces=None, loop=self.loop,
+                session=None
+            )
+        )
+
```

Comment:
Check the global variable: `self.visited_urls`, `self.meta` , `self.new_urls`etc.